code/game/objects/structures/barsign.dm

```diff
@@ -47,68 +52,141 @@
 /obj/structure/sign/barsign/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
 	switch(damage_type)
 		if(BRUTE)
-			playsound(src.loc, 'sound/effects/glasshit.ogg', 75, TRUE)
+			playsound(loc, 'sound/effects/glasshit.ogg', 75, TRUE)
 		if(BURN)
-			playsound(src.loc, 'sound/items/welder.ogg', 100, TRUE)
-
-/obj/structure/sign/barsign/attack_ai(mob/user as mob)
-	return src.attack_hand(user)
+			playsound(loc, 'sound/items/welder.ogg', 100, TRUE)
 
+/obj/structure/sign/barsign/attack_ai(mob/user)
+	return attack_hand(user)
 
-
-/obj/structure/sign/barsign/attack_hand(mob/user as mob)
-	if(!src.allowed(user))
-		to_chat(user, "<span class = 'info'>Access denied.</span>")
-		return
-	if(broken)
-		to_chat(user, "<span class ='danger'>The controls seem unresponsive.</span>")
+/obj/structure/sign/barsign/attack_hand(mob/user)
+	if(panel_open)
+		if(allowed(user))
+			if(broken)
+				to_chat(user, "<span class='danger'>The controls seem unresponsive.</span>")
+				return
+			else
+				pick_sign()
+				to_chat(user, "<span class='notice'>You set the barsign and close the maintenance panel.</span>")
+				panel_open = FALSE
+				return
+		else
+			to_chat(user, "<span class='info'>Access denied.</span>")
+			return
+	else
+		to_chat(user, "<span class='info'>The maintenance panel is currently closed.</span>")
 		return
-	pick_sign()
-
 
 
-
-/obj/structure/sign/barsign/attackby(var/obj/item/I, var/mob/user)
-	if( istype(I, /obj/item/screwdriver))
-		if(!panel_open)
-			to_chat(user, "<span class='notice'>You open the maintenance panel.</span>")
-			set_sign(new /datum/barsign/hiddensigns/signoff)
-			panel_open = 1
-		else
-			to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
-			if(!broken && !emagged)
-				set_sign(pick(barsigns))
-			else if(emagged)
-				set_sign(new /datum/barsign/hiddensigns/syndibarsign)
-			else
-				set_sign(new /datum/barsign/hiddensigns/empbarsign)
-			panel_open = 0
-
+/obj/structure/sign/barsign/attackby(obj/item/I, mob/user)
 	if(istype(I, /obj/item/stack/cable_coil) && panel_open)
 		var/obj/item/stack/cable_coil/C = I
-		if(emagged) //Emagged, not broken by EMP
+		if(broken && emagged) //Emagged, not broken by EMP
 			to_chat(user, "<span class='warning'>Sign has been damaged beyond repair!</span>")
 			return
-		else if(!broken)
+		else if(broken && !emagged)
+			if(C.use(2))
+				to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
+				broken = FALSE
+			else
+				to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+		else if(!broken && !emagged)
 			to_chat(user, "<span class='warning'>This sign is functioning properly!</span>")
 			return
-
-		if(C.use(2))
-			to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
-			broken = 0
-		else
-			to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+	if(isscrewdriver(I) || iswrench(I))
+		return	// No damage.
```

Comment:
IIRC you can get rid of these checks and just `return TRUE` in the tool-specific `act`s to prevent attacking when needed, but I'm not 100% sure